src/ipc-handlers.ts

```diff
@@ -1711,27 +1712,59 @@ export function comparePaths( event: IpcMainInvokeEvent, path1: string, path2: s
 	return arePathsEqual( path1, path2 );
 }
 
-export async function listWpContentFolders(
+export async function listLocalFileTree(
 	_event: Electron.IpcMainInvokeEvent,
 	siteId: string,
-	subdir: GranularSyncFolders
-): Promise< { name: string; type: 'file' | 'folder' }[] > {
+	path: string,
+	maxDepth: number = 3,
+	currentDepth: number = 0
+): Promise< RawDirectoryEntry[] > {
 	const server = SiteServer.get( siteId );
 	if ( ! server ) throw new Error( 'Site not found' );
-	const wpContentPath = nodePath.join( server.details.path, 'wp-content', subdir );
+
+	const fullPath = nodePath.join( server.details.path, path );
 
 	try {
-		const entries = await fs.promises.readdir( wpContentPath, { withFileTypes: true } );
-		return entries
-			.map( ( e ) => ( {
-				name: e.name.toString(),
-				type: e.isDirectory() ? ( 'folder' as const ) : ( 'file' as const ),
-				hidden: e.name.startsWith( '.' ),
-			} ) )
-			.filter( ( entry: { name: string; hidden: boolean } ) => {
-				return ! entry.hidden;
-			} );
+		const entries = await fs.promises.readdir( fullPath, { withFileTypes: true } );
+		const result = [];
+
+		for ( const entry of entries ) {
+			if ( shouldExcludeFromSync( entry.name ) ) {
+				continue;
+			}
+
+			const isDirectory = entry.isDirectory();
+			const itemPath = nodePath.join( path, entry.name ).replace( /\\/g, '/' );
+			const fullItemPath = isDirectory ? itemPath + '/' : itemPath;
+
+			const fileNode: RawDirectoryEntry = {
```

Comment:
```suggestion
			const directoryEntry: RawDirectoryEntry = {
```
Nit, but let's rename the variable, too, if it can represent both files and directories.

Reply:
Sounds good, adjusted in https://github.com/Automattic/studio/pull/1940/commits/b3db695fe4a22b92160a0120befb1764207e738f